main_test.go

```diff
@@ -1,9 +1,54 @@
 package main
 
-import ()
+import (
+	"io/ioutil"
+	"testing"
+)
 
 func setup() {
 }
 
 func teardown() {
 }
+
+func TestAppWithoutApiKey(t *testing.T) {
+	app := buildApp()
+	app.Writer = ioutil.Discard
+
+	// test without other global flags
+	args := []string{"doctl"}
+	err := app.Run(args)
+
+	expected := "Must provide API Key via DIGITALOCEAN_API_KEY environment variable or via CLI argument."
+	if err.Error() != expected {
+		t.Errorf("err = %v, expected %v", err.Error(), expected)
+	}
+
+	// test with other global flags
+	args = []string{"doctl", "--version"}
```

Comment:
this could be refactored in a test table:

```go
tests := []struct{
    args []string
    wantErr string
}{
   {
        args: []string{"doctl", "--version"},
        wantErr "derpderpderp",
   },
   // derp derp derp
}

for _, tt := range tests {
    err := app.Run(tt.args)
    if err.Error() != tt.wantErr { // need more to check for nil
        tt.Fatalf("app.Run(%v) = %#v, want %#v", tt.args, err, tt.wantErr)
    }
}
```